conans/tools.py

```diff
@@ -183,15 +183,27 @@ def vcvars_command(settings):
                                  % (existing_version, compiler_version))
     else:
         env_var = "vs%s0comntools" % compiler_version
+
+        vs_path = None
+        if env_var == 'vs150comntools' and not env_var in os.environ:
+            try:
```

Comment:
This is cool.
I would like to have this functionality extracted, so it can be called in the "detect" initial routine to define computer defaults. Also, it would be great to cache the result, to avoid calling vswhere many times. Could you do it yourself, or do you want me to help with that?